Implements (quite) the same as BR2_LINUX_COPYTO for U-Boot.

Signed-off-by: Julien Boibessot <julien.boibessot@armadeus.com>

Index: buildroot/boot/u-boot/Config.in
===================================================================
--- buildroot.orig/boot/u-boot/Config.in	2010-06-17 16:45:13.000000000 +0200
+++ buildroot/boot/u-boot/Config.in	2010-06-17 16:46:45.000000000 +0200
@@ -79,6 +79,12 @@
 	help
 	  Install fw_printenv / fw_setenv tools in target.
 
+config BR2_TARGET_UBOOT_COPYTO
+	string "Copy binary image to <dir>..."
+	default ""
+	help
+	  Copy U-Boot binary image to secondary location.
+
 menuconfig BR2_TARGET_UBOOT_NETWORK
 	bool "Network Settings"
 	default y
Index: buildroot/boot/u-boot/u-boot.mk
===================================================================
--- buildroot.orig/boot/u-boot/u-boot.mk	2010-06-17 16:45:13.000000000 +0200
+++ buildroot/target/u-boot/u-boot.mk	2010-06-17 16:46:45.000000000 +0200
@@ -54,6 +54,11 @@
 U_BOOT_TARGETS+=$(TARGET_DIR)/usr/sbin/fw_printenv
 endif
 
+ifneq ($(call qstrip,$(BR2_TARGET_UBOOT_COPYTO)),)
+U_BOOT_COPYTO:=$(call qstrip,$(BR2_TARGET_UBOOT_COPYTO))
+U_BOOT_TARGETS+=$(U_BOOT_COPYTO)/$(U_BOOT_TARGET_BIN)
+endif
+
 U_BOOT_CONFIGURE_OPTS += CONFIG_NOSOFTFLOAT=1
 
 # Define a helper function
@@ -138,6 +143,12 @@
 	rm -f $(BINARIES_DIR)/$(U_BOOT_BIN)
 	cp -dpf $(U_BOOT_DIR)/$(U_BOOT_BIN) $(BINARIES_DIR)/
 
+ifneq ($(U_BOOT_COPYTO),)
+$(U_BOOT_COPYTO)/$(U_BOOT_TARGET_BIN): $(U_BOOT_DIR)/$(U_BOOT_BIN)
+	mkdir -p $(U_BOOT_COPYTO)
+	cp -dpf $< $@
+endif
+
 # Build just mkimage for the host. It might have already been built by
 # the U-Boot build procedure, but mkimage may also be needed even if
 # U-Boot isn't selected in the configuration, to generate a kernel
